src/components/users/Accounts.jsx

```diff
@@ -63,14 +93,39 @@ const CreateAccount = ({
     setAccounts,
 }) => {
     const [fullName, setFullName] = useState(accounts.fullName);
+    const [_userAccount, _setUserAccount] = useState(accounts.userAccount);
     const [userAccount, setUserAccount] = useState(accounts.userAccount);
+    const [userAccountInvalidHint, setUserAccountInvalidHint] = useState("");
+    const [isUserAccountValid, setIsUserAccountValid] = useState(null);
     const [password, setPassword] = useState(accounts.password);
     const [confirmPassword, setConfirmPassword] = useState(accounts.confirmPassword);
     const [isPasswordValid, setIsPasswordValid] = useState(false);
 
     useEffect(() => {
-        setIsUserValid(isPasswordValid && userAccount.length > 0);
-    }, [setIsUserValid, isPasswordValid, userAccount]);
+        debounce(300, () => setUserAccount(_userAccount))();
+    }, [_userAccount, setUserAccount]);
+
+    useEffect(() => {
+        setIsUserValid(isPasswordValid && isUserAccountValid);
+    }, [setIsUserValid, isPasswordValid, isUserAccountValid]);
+
+    useEffect(() => {
+        let valid = true;
+        setUserAccountInvalidHint("");
+        if (userAccount.length === 0) {
+            valid = null;
+        } else if (userAccount.length > 32) {
+            valid = false;
+            setUserAccountInvalidHint(_("The user name is too long"));
```

Comment:
Order of preference for the account name is:

1. Account name (doesn't use "user"; this is good, as "user" is bad to show in interfaces)
2. Username (the actual word, according to English dictionaries)
3. User name (an uncommon variant, according to English dictionaries)

As both GNOME and KDE talk about "user" accounts and have a "name" field, I guess keeping with "User name" is probably OK enough for consistency, even if it's not ideal.

Therefore, this is fine.

Reply:
On the other hand, it would be good to mention what the cutoff is.

```suggestion
            setUserAccountInvalidHint(_("User names must contain fewer than 33 characters"));
```

Reply:
Updated